src/PolyhedralGeometry/Polyhedron/properties.jl

```diff
@@ -1331,6 +1331,137 @@ false
 """
 is_fulldimensional(P::Polyhedron) = pm_object(P).FULL_DIM::Bool
 
+@doc raw"""
+    is_johnson_solid(P::Polyhedron)
+
+Check whether `P` is a Johnson solid, i.e., a $3$-dimensional polytope with regular faces that is not vertex transitive.
+
+See also [`johnson_solid`](@ref).
+
+!!! note
+    This will only recognize algebraically precise solids, i.e. no solids with approximate coordinates.
+
+# Examples
+```
+julia> J = johnson_solid(37)
+Polytope in ambient dimension 3 with EmbeddedAbsSimpleNumFieldElem type coefficients
+
+julia> is_johnson_solid(J) 
+true
+```
+"""
+is_johnson_solid(P::Polyhedron) = _is_3d_pol_reg_faces(P) && !is_vertex_transitive(P)
+
+@doc raw"""
+    is_archimedean_solid(P::Polyhedron)
+
+Check whether `P` is an Archimedean solid, i.e., a $3$-dimensional vertex transitive polytope with regular faces.
+
+See also [`archimedean_solid`](@ref).
+
+!!! note
+    This will only recognize algebraically precise solids, i.e. no solids with approximate coordinates.
+
+# Examples
+```jldoctest
+julia> TO = archimedean_solid("truncated_octahedron")
+Polytope in ambient dimension 3
+
+julia> is_archimedean_solid(TO)
+true
+
+julia> T = tetrahedron()
+Polytope in ambient dimension 3
+
+julia> is_archimedean_solid(T)
+false
+```
+"""
+is_archimedean_solid(P::Polyhedron) = _is_3d_pol_reg_faces(P) && !_has_equal_faces(P) && is_vertex_transitive(P)
+
+@doc raw"""
+    is_platonic_solid(P::Polyhedron)
+
+Check whether `P` is a Platonic solid.
+
+See also [`platonic_solid`](@ref).
+
+!!! note
+    This will only recognize algebraically precise solids, i.e. no solids with approximate coordinates.
+
+# Examples
+```jldoctest
+julia> is_platonic_solid(cube(3))
+true
+```
+"""
+is_platonic_solid(P::Polyhedron) = _is_3d_pol_reg_faces(P) && _has_equal_faces(P) && is_vertex_transitive(P)
+
+function _is_3d_pol_reg_faces(P::Polyhedron)
+  # dimension
+  dim(P) == 3 || return false
+
+  # constant edge length
+  pedges = faces(P, 1)
+
+  edgelength = nothing
+  let edge = pedges[1]
+    v = vertices(edge)
+    edgelength = _squared_distance(v[1], v[2])
+  end
+
+  for edge in pedges[2:end]
+    v = vertices(edge)
+    _squared_distance(v[1], v[2]) == edgelength || return false
+  end
+
+  pfaces = faces(P, 2)
+
+  # face vertices on circle
+  for face in pfaces
+    n = n_vertices(face)
+    if n >= 4
+      fverts = vertices(face)
+      m = sum(fverts)//n
+      dist = nothing
+      let v = fverts[1]
+        dist = _squared_distance(v, m)
+      end
```

Comment:
```suggestion
      dist = _squared_distance(fverts[1], m)
```
Using the let block when v is just used once is not really necessary and I don't think it increases readability.